grid_map_filters/include/EigenLab/EigenLab.hpp

```diff
@@ -0,0 +1,2847 @@
+  inline const Derived & local() const {return mLocal;}
+
+  // Is mapped data local?
+  inline bool isLocal() const {return mIsLocal;}
+
+  // Set mapped data to point to local data.
+  inline void mapLocal()
+  {
+    new(&mShared) Eigen::Map<Derived>(mLocal.data(), mLocal.rows(), mLocal.cols());
+    mIsLocal = true;
+  }
+
+  //  Copy shared data to local data (if needed).
+  inline void copySharedToLocal() {if (!isLocal()) {mLocal = mShared; mapLocal();}}
+
+  // Set local data.
+  Value()
+  : mLocal(1, 1), mShared(mLocal.data(), mLocal.rows(), mLocal.cols()), mIsLocal(true)
+  {
+  }
+  explicit Value(const typename Derived::Scalar s)
+  : mLocal(Derived::Constant(1, 1, s)), mShared(
+      mLocal.data(), mLocal.rows(), mLocal.cols()), mIsLocal(true)
+  {
+  }
+  explicit Value(const Derived & mat)
+  : mLocal(mat), mShared(mLocal.data(), mLocal.rows(), mLocal.cols()),
+    mIsLocal(true)
+  {
+  }
+  inline void setLocal(const typename Derived::Scalar s)
+  {
+    mLocal = Derived::Constant(1, 1, s); mapLocal();
+  }
+  inline void setLocal(const Eigen::MatrixBase<Derived> & mat) {mLocal = mat; mapLocal();}
+  inline void setLocal(const Value & val) {mLocal = val.matrix(); mapLocal();}
+  inline void setLocal(const typename Derived::Scalar * data, size_t rows = 1, size_t cols = 1)
+  {
+    setShared(data, rows, cols); copySharedToLocal();
+  }
+  inline Value & operator=(const typename Derived::Scalar s) {setLocal(s); return *this;}
+  inline Value & operator=(const Derived & mat) {setLocal(mat); return *this;}
+
+  // Set shared data.
+  explicit Value(const typename Derived::Scalar * data, size_t rows = 1, size_t cols = 1)
+  : mShared(
+      const_cast<typename Derived::Scalar *>(data), rows, cols), mIsLocal(false)
+  {
+  }
+  inline void setShared(const typename Derived::Scalar * data, size_t rows = 1, size_t cols = 1)
+  {
+    new(&mShared) Eigen::Map<Derived>(const_cast<typename Derived::Scalar *>(data), rows, cols);
+    mIsLocal = false;
+  }
+  inline void setShared(const Derived & mat) {setShared(mat.data(), mat.rows(), mat.cols());}
+  inline void setShared(const Value & val)
+  {
+    setShared(val.matrix().data(), val.matrix().rows(), val.matrix().cols());
+  }
+
+  // Set to local or shared data dependent on whether val
+  // maps its own local data or some other shared data.
+  Value(const Value & val)
+  : mLocal(1, 1), mShared(mLocal.data(), mLocal.rows(), mLocal.cols())
+  {
+    (*this) = val;
+  }
+  inline Value & operator=(const Value & val)
+  {
+    if (val.isLocal()) {
+      setLocal(val);
+    } else {
+      setShared(val);
+    } return *this;
+  }
+};
+typedef Value<Eigen::MatrixXd> ValueXd;
+typedef Value<Eigen::MatrixXf> ValueXf;
+typedef Value<Eigen::MatrixXi> ValueXi;
+
+// check if a class has a comparison operator (ie. std::complex does not)
+template<typename T>
+struct has_operator_lt_impl
+{
+  template<class U>
+  static auto test(U *)->decltype(std::declval<U>() < std::declval<U>());
```

Comment:
Not sure what is the purpose/intention of this, [deep-code CI](https://www.deepcode.ai/app/gh/ANYbotics/grid_map/07c8429b7bcabe182b9f2c203b2f10d3de583f3c/ANYbotics/grid_map/daa32ac20b6b7da991d4be7de85c424971a11c86/pr/_/%2Fgrid_map_filters%2Finclude%2FEigenLab%2FEigenLab.hpp/cpp%2Fdc%2FCopyPasteError/165/code) is complaining about it. Any suggestions?

One a slightly different tangent I think its worth refactoring this file at some point down the line.

Reply:
I'm not familiar with deep-code, whenever I try to find out more, it asks for a bunch of log ins I'm not comfortable giving it permissions for

Reply:
@samuelba do you know more about the deep-code bot?

Reply:
Can you add

`//  deepcode ignore CopyPasteError: We will not change third party code yet.`

?